### PR TITLE
remove small remnants of memory_ballast. OTL-2703

### PR DESCRIPTION
**Pull request by samiura**
**Description:** Removing ineffective code which used to be used for` memory_ballast`

**Link to Splunk idea:** #4548

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Review by atoulme — APPROVED** [review]

File: internal/settings/settings.go [hunk 1]
```diff
@@ -523,21 +518,6 @@ func envVarAsInt(env string) int {
 	return val
 }
 
-// Validate and set the memory ballast.
-// Note this will eventually be removed and here only for backward compatibility. Softlimit/GOMemlimit sets memory limit.
-func setMemoryBallast(memTotalSizeMiB int) int {
-	ballastSize := memTotalSizeMiB * DefaultMemoryBallastPercentage / 100
```
Comment (dmitryax) [inline internal/settings/settings.go:529]: `DefaultMemoryBallastPercentage` also can be removed, right?

File: internal/settings/settings.go [hunk 2]
```diff
@@ -523,21 +518,6 @@ func envVarAsInt(env string) int {
...
-	// Check if the memory ballast is specified via the env var, if so, validate and set properly.
-	if os.Getenv(BallastEnvVar) != "" {
```
Comment (dmitryax) [inline internal/settings/settings.go:531]: Same for BallastEnvVar

Comment (samiura) [reply internal/settings/settings.go:531]: done

**Review by jvoravong — APPROVED** [review]

**Review by pjanotti — APPROVED** [review]

File: internal/settings/settings.go [hunk 3]
```diff
@@ -371,16 +370,11 @@ func checkRuntimeParams(settings *Settings) error {
 		}
 	}
 
-	ballastSize := setMemoryBallast(memTotalSize)
-	memLimit, err := setMemoryLimit(memTotalSize)
+	_, err := setMemoryLimit(memTotalSize)
```
Comment (dmitryax) [inline internal/settings/settings.go:373]: `setMemoryLimit` should also be updated to not return a value if it's not needed anymore

**Pull request merged**